### PR TITLE
fixed issues aoh & task list changes

### PR DESCRIPTION
**Pull request by praneethhm**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] the `keep-helm` label has been added, if the helm release should be persisted after a successful build

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/FPET-682

### Change description ###
fixed issues aoh & task list changes

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/prl/enums/EventErrorsEnum.java [hunk 1]
```diff
@@ -8,9 +8,9 @@
 public enum EventErrorsEnum {
 
     CHILD_DETAILS_ERROR("Add people to the case details to"),
```
Comment (priteshkr) [inline src/main/java/uk/gov/hmcts/reform/prl/enums/EventErrorsEnum.java:10]: Remove "to" from the end

File: src/main/java/uk/gov/hmcts/reform/prl/enums/EventErrorsEnum.java [hunk 2]
```diff
@@ -21,9 +21,9 @@ public enum EventErrorsEnum {
     WELSH_LANGUAGE_ERROR("Add details about welsh language requirements"),
     OTHER_PROCEEDINGS_ERROR("Add details about other proceedings"),
     ALLEGATIONS_OF_HARM_ERROR("Add details about allegations of harm"),
-    OTHER_CHILDREN_NOT_PART_OF_THE_APPLICATION_ERROR("Add people to the case details to"),
-    CHILDREN_AND_APPLICANTS_ERROR("Add people to the case details to"),
-    CHILDREN_AND_RESPONDENTS_ERROR("Add people to the case details to"),
+    OTHER_CHILDREN_NOT_PART_OF_THE_APPLICATION_ERROR("Add people to the case details"),
```
Comment (priteshkr) [inline src/main/java/uk/gov/hmcts/reform/prl/enums/EventErrorsEnum.java:24]: The error does not look relevant according to the relevant event

File: src/main/java/uk/gov/hmcts/reform/prl/enums/EventErrorsEnum.java [hunk 3]
```diff
@@ -21,9 +21,9 @@ public enum EventErrorsEnum {
...
+    CHILDREN_AND_APPLICANTS_ERROR("Add people to the case details"),
+    CHILDREN_AND_RESPONDENTS_ERROR("Add people to the case details"),
     CHILDREN_AND_OTHER_PEOPLE_IN_THIS_APPLICATION_ERROR("Add people to the case details to"),
```
Comment (priteshkr) [inline src/main/java/uk/gov/hmcts/reform/prl/enums/EventErrorsEnum.java:27]: The error does not look relevant according to the relevant event

File: src/main/java/uk/gov/hmcts/reform/prl/services/validators/SubmitAndPayChecker.java [hunk 4]
```diff
@@ -59,43 +59,24 @@ public boolean isStarted(CaseData caseData) {
...
-            }
-        }
-
         EnumMap<Event, EventChecker> optionalEvents = new EnumMap<>(Event.class);
         if (TASK_LIST_VERSION_V2.equalsIgnoreCase(caseData.getTaskListVersion())) {
             optionalEvents.put(
-                OTHER_CHILDREN_NOT_PART_OF_THE_APPLICATION,
-                eventsChecker.getOtherChildrenNotPartOfTheApplicationChecker()
+                    OTHER_CHILDREN_NOT_PART_OF_THE_APPLICATION,
+                    eventsChecker.getOtherChildrenNotPartOfTheApplicationChecker()
```
Comment (priteshkr) [inline src/main/java/uk/gov/hmcts/reform/prl/services/validators/SubmitAndPayChecker.java:69]: Other child not part of the application is mandatory ?

File: src/main/resources/C100_Dummy_Draft_CaseDetails.json [hunk 5]
```diff
@@ -8,673 +8,40 @@
...
...busercontent.com/hmcts/prl-cos-api/master/resources/not-started.png' title='Not started'/>\n\n<hr class='govuk-!-margin-top-3 govuk-!-margin-bottom-2'/>\n\n<a href='/cases/case-details/${[CASE_REFERENCE]}/trigger/welshLanguageRequirements/welshLanguageRequirements1'>Welsh language requirements</a><img align='right' height='25px' src='https://raw.githubusercontent.com/hmcts/prl-cos-api/master/resources/finished.png' title='Finished'/>\n\n<hr class='govuk-!-margin-top-3 govuk-!-margin-bottom-2'/>\n\n<br/>\n\n## View PDF application\n\n<hr class='govuk-!-margin-top-3 govuk-!-margin-bottom-2'/>\n\n<a href='/cases/case-details/${[CASE_REFERENCE]}/trigger/viewPdfDocument/viewPdfDocument1'>View PDF application</a>\n\n<hr class='govuk-!-margin-top-3 govuk-!-margin-bottom-2'/>\n\n<br/>\n\n## Submit and pay\n\n<hr class='govuk-!-margin-top-3 govuk-!-margin-bottom-2'/>\n\n<a href='/cases/case-details/${[CASE_REFERENCE]}/trigger/submitAndPay/submitAndPay1'>Submit and pay</a><img align='right' height='25px' src='https://raw.githubusercontent.com/hmcts/prl-cos-api/master/resources/not-started.png' title='Not started yet'/>\n\n<hr class='govuk-!-margin-top-3 govuk-!-margin-bottom-2'/>\n\n</div>",
```
Comment (priteshkr) [inline src/main/resources/C100_Dummy_Draft_CaseDetails.json:12]: We have to keep the applicant details, respondent details and child details same as last time. The people including service team are used to old applicant ,respondent and child details.

File: src/main/resources/C100_Dummy_Gatekeeping_CaseDetails.json [hunk 6]
```diff
@@ -8,2098 +8,59 @@
...
...busercontent.com/hmcts/prl-cos-api/master/resources/not-started.png' title='Not started'/>\n\n<hr class='govuk-!-margin-top-3 govuk-!-margin-bottom-2'/>\n\n<a href='/cases/case-details/${[CASE_REFERENCE]}/trigger/welshLanguageRequirements/welshLanguageRequirements1'>Welsh language requirements</a><img align='right' height='25px' src='https://raw.githubusercontent.com/hmcts/prl-cos-api/master/resources/finished.png' title='Finished'/>\n\n<hr class='govuk-!-margin-top-3 govuk-!-margin-bottom-2'/>\n\n<br/>\n\n## View PDF application\n\n<hr class='govuk-!-margin-top-3 govuk-!-margin-bottom-2'/>\n\n<a href='/cases/case-details/${[CASE_REFERENCE]}/trigger/viewPdfDocument/viewPdfDocument1'>View PDF application</a>\n\n<hr class='govuk-!-margin-top-3 govuk-!-margin-bottom-2'/>\n\n<br/>\n\n## Submit and pay\n\n<hr class='govuk-!-margin-top-3 govuk-!-margin-bottom-2'/>\n\n<a href='/cases/case-details/${[CASE_REFERENCE]}/trigger/submitAndPay/submitAndPay1'>Submit and pay</a><img align='right' height='25px' src='https://raw.githubusercontent.com/hmcts/prl-cos-api/master/resources/not-started.png' title='Not started yet'/>\n\n<hr class='govuk-!-margin-top-3 govuk-!-margin-bottom-2'/>\n\n</div>",
```
Comment (priteshkr) [inline src/main/resources/C100_Dummy_Gatekeeping_CaseDetails.json:13]: We have to keep the applicant details, respondent details and child details same as last time. The people including service team are used to old applicant ,respondent and child details.

File: src/main/java/uk/gov/hmcts/reform/prl/rpa/mappers/AllegationsOfHarmRevisedMapper.java [hunk 7]
```diff
@@ -175,11 +175,41 @@ private JsonArray mapChildAbuseBehaviours(AllegationOfHarmRevised allegationOfHa
             return JsonValue.EMPTY_JSON_ARRAY;
         }
         List<ChildAbuse> childAbuseBehavioursList = new ArrayList<>();
-        childPhysicalAbuse.ifPresent(childAbuseBehavioursList::add);
-        childPsychologicalAbuse.ifPresent(childAbuseBehavioursList::add);
-        childEmotionalAbuse.ifPresent(childAbuseBehavioursList::add);
-        childSexualAbuse.ifPresent(childAbuseBehavioursList::add);
-        childFinancialAbuse.ifPresent(childAbuseBehavioursList::add);
+
```
Comment (priteshkr) [inline src/main/java/uk/gov/hmcts/reform/prl/rpa/mappers/AllegationsOfHarmRevisedMapper.java:178]: Why have made this change ? what was the issue raised for this ?

Comment (priteshkr) [reply src/main/java/uk/gov/hmcts/reform/prl/rpa/mappers/AllegationsOfHarmRevisedMapper.java:178]: Use Objects.nonNull

File: src/main/java/uk/gov/hmcts/reform/prl/services/validators/SubmitAndPayChecker.java (hunk 4 above)
Comment (priteshkr) [reply src/main/java/uk/gov/hmcts/reform/prl/services/validators/SubmitAndPayChecker.java:69]: Ignore this comment. As we can select "No".

**Review by priteshkr — APPROVED** [review]

**Review by sairamkantheti1 — APPROVED** [review]

**Pull request merged**